src/main/java/net/dries007/tfc/objects/items/metal/ItemMetalBucket.java

```diff
@@ -124,6 +124,10 @@ else if (fluid.getBlock() instanceof BlockLiquid)
                             }
                         }
                         worldIn.playSound(null, pos, SoundEvents.ITEM_BUCKET_EMPTY, SoundCategory.PLAYERS, 1.0F, 1.0F);
+                        if (stack.hasTagCompound() && stack.getTagCompound().isEmpty())
+                        {
+                            stack.setTagCompound(null);
+                        }
```

Comment:
So I understand the issue that this is trying to solve, but this is a really hack and slash way of getting around it and doesn't resolve the root issue. There are still plenty of situations where the NBT tag would still apply. As I mentioned in the issue, we need to tackle the root problem, namely, the situations in `FluidHandlerItemStackSimple` which saves the fluid to NBT, is not the same assumption that we operate under (that the fluid is never saved to NBT). For consistency with the rest of the mod we should go with the latter, but for simplicity it might be better to go with the former.